wagtailmedia/static/wagtailmedia/js/media-chooser-modal.js

```diff
@@ -0,0 +1,144 @@
+MEDIA_CHOOSER_MODAL_ONLOAD_HANDLERS = {
+    'chooser': function(modal, jsonData) {
+        var searchUrl = $('form.media-search', modal.body).attr('action');
+
+        /* currentTag stores the tag currently being filtered on, so that we can
+        preserve this when paginating */
+        var currentTag;
+
+        function ajaxifyLinks (context) {
+            $('a.media-choice', context).on('click', function() {
+                modal.loadUrl(this.href);
+                return false;
```

Comment:
@thibaudcolas I found that this doesn't prevent the link from redirecting to the media file URL (at least in Firefox). I'll add a patch to the PR to fix this behaviour.